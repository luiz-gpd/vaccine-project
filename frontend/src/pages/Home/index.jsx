import React from 'react'
import ClayButton from '@clayui/button'
import Page from '../../components/Page'

const Home = ( {history} ) => {
    
    console.log(history.push);

    return (
        <div id="container">
        <Page className="mt-4">
            <h1 className="welcome">Bem Vindo!</h1>
            <h2 className="mt-4">Devemos nos cuidar sempre!</h2>
            <h2>Tomar a vacina é um cuidado e dever de todos!</h2>

            <ClayButton data-testid="homeButton" className="mt-4"
            onClick={() => {history.push('/user')}}>Agendar Vacinação!</ClayButton>

        </Page>
        </div>
    )
}

export default Home;