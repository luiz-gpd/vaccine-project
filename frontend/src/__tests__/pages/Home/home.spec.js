import Home from "../../../pages/Home"
import { createMemoryHistory } from 'history'
import { render, fireEvent, waitFor } from "@testing-library/react"

describe("Home Page", () => {
    const history = createMemoryHistory;

    it("renders", () => {
        const { asFragment } = render(<Home
            history={history}
        />);

        expect(asFragment()).toMatchSnapshot();
    });
    it("Should submit when clicking on the button", async () => {
        const history = createMemoryHistory;
        
        const { getByTestId } = render(<Home
            history={history}
        />);
    
        // const homeButton = await waitFor(() => getByTestId('homeButton'))
        // fireEvent.click(homeButton)
    });
});