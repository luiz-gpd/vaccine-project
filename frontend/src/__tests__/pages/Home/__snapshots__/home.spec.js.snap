// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page renders 1`] = `
<DocumentFragment>
  <div
    id="container"
  >
    <div
      class="mt-4 container"
    >
      <div
        class="row justify-content-between"
      >
        <div
          class="col"
        >
          <h1 />
        </div>
        <div
          class="col-3"
        >
          <img
            alt="Logo do site"
            src="VaciCuidar_logo.png"
            width="250px"
          />
        </div>
      </div>
      <h1
        class="welcome"
      >
        Bem Vindo!
      </h1>
      <h2
        class="mt-4"
      >
        Devemos nos cuidar sempre!
      </h2>
      <h2>
        Tomar a vacina é um cuidado e dever de todos!
      </h2>
      <button
        class="mt-4 btn btn-primary"
        data-testid="homeButton"
        type="button"
      >
        Agendar Vacinação!
      </button>
    </div>
  </div>
</DocumentFragment>
`;
